__device-tests__/utils.js

```diff
@@ -157,3 +157,98 @@ export async function takeScreenshot( { withoutKeyboard, crop } = {} ) {
 	const resizedImage = await image.resize( widthSize, jimp.AUTO );
 	return resizedImage.getBufferAsync( jimp.MIME_PNG );
 }
+
+/**
+ * Generates the appropriate URL to fetch the theme data based on the provided theme details.
+ *
+ * @param {Object}  theme                  Object containing details about the theme.
+ * @param {string}  theme.name             The name of the theme.
+ * @param {boolean} theme.isWordPressTheme Flag indicating whether the theme is a WordPress theme. Defaults to false.
+ * @return {string} The URL from which the theme data can be fetched.
+ */
+function getThemeLink( { name, isWordPressTheme = false } ) {
+	if ( isWordPressTheme ) {
+		return `https://raw.githubusercontent.com/WordPress/${ name }/trunk/theme.json`;
+	}
+
+	return `https://raw.githubusercontent.com/Automattic/themes/trunk/${ name }/theme.json`;
+}
+
+/**
+ * Fetches and processes the theme data from the provided theme details.
+ *
+ * @param {Object}  theme                  Object containing details about the theme.
+ * @param {string}  theme.name             The name of the theme.
+ * @param {boolean} theme.isWordPressTheme Flag indicating whether the theme is a WordPress theme. Defaults to false.
+ * @return {Promise<Object>} A promise that resolves to a stringified JSON object containing the theme data.
+ * @throws Will throw an error if the fetch operation fails.
+ */
+export async function fetchTheme( theme ) {
+	const themeJSONLink = getThemeLink( theme );
+
+	return await fetch( themeJSONLink )
+		.then( ( response ) => response.json() )
+		.then( ( data ) => {
+			const rawFeatures = data?.settings;
+			const rawStyles = data?.styles;
+			const supportedBlocks = [
+				'core/paragraph',
+				'core/button',
+				'core/quote',
+				'core/pullquote',
+				'core/quote',
+				'core/search',
```

Comment:
The issue is that if we allow passing more global styles for other blocks that we do support but don't support rendering their custom styles, it increases the length of the JSON that we send through the launch app params. On Android, if it's too long it gets cut off 😅

Reply:
> On Android, if it's too long it gets cut off 😅

Yep, this might be a limitation in the future. I'm wondering if we could create a temporal JSON file with all the data and pass it in the launch arguments instead 🤔.

Reply:
> Yep, this might be a limitation in the future. I'm wondering if we could create a temporal JSON file with all the data and pass it in the launch arguments instead 🤔.

What do you mean by passing the data in the launch arguments? I thought this is what we are doing [in this code](https://github.com/WordPress/gutenberg/blob/trunk/packages/react-native-editor/__device-tests__/helpers/utils.js#L758-L768) I haven't seen another way to pass data through the launch functionality 😅

Reply:
The change I was thinking of is that instead of passing the raw data in the launch arguments, as we currently do, create a temporal JSON file with all the data and pass its file path in the launch arguments. This way the app would fetch all the data from a file, which I assume we won't have limitations in terms of the content length. The tricky part is that this JSON file should be accessible by the app in SauceLabs 🤔.

In any case, I think this approach would need further exploration. Since we don't have yet the need to pass complex structures, let's omit it for now.